homeassistant/config.py

```diff
@@ -669,6 +674,39 @@ def async_process_component_config(hass, config, domain):
     return None
 
 
+async def async_check_ha_config_file_new(hass, config_path=None):
+    """Check if Home Assistant configuration file is valid."""
+    if not config_path:
+        config_path = hass.config.config_dir
+
+    try:
+        config = load_yaml_config_file(config_path)
+    finally:
+        clear_secret_cache()
+
+    # Validate and extract core config
+    core_config = CORE_CONFIG_SCHEMA(config.get(CONF_CORE, {}))
+    del config[CONF_CORE]
+
+    # Merge packages
+    merge_packages_config(
+        config, core_config.get(CONF_PACKAGES, {}))
+
+    # Filter out the repeating and common config section [homeassistant]
+    components = set(key.split(' ')[0] for key in config.keys())
+    print(components)
+
+    # To ADD config entries?
+    # hass.config_entries = config_entries.ConfigEntries(hass, config)
+    # yield from hass.config_entries.async_load()
+    # components.update(hass.config_entries.async_domains())
```

Comment:
Config entries are a complete standalone configuration channel from normal config. They also don't have validation as they are read and written by code.